### PR TITLE
Userspace test framework overhaul (part 1)

### PR DESCRIPTION
**Pull request by cahirwpz**
* add TEST_ADD macro to define new tests
* add set of macros in utest.h to perform checks
* shorten duration of some tests that used sleep/nanosleep
* add ualarm implementation from NetBSD
* allow klog to be dumped to file
* each user test logs start time
* launch: correctly report test timeout
* launch: print output from console on test run failure or timeout

### CODE REVIEW

File: bin/utest/cred.c [hunk 1]
```diff
@@ -1,155 +1,130 @@
...
-int test_get_set_groups(void) {
+TEST_ADD(get_set_groups) {
   gid_t rgrp[NGROUPS_MAX], gidset[NGROUPS_MAX] = {0, 1, 2, 3, 4, 5};
-  int r, ngroups = 6;
+  const int ngroups = 6;
   uid_t euid;
 
   /* check if we are a root at start */
   getresuid(NULL, &euid, NULL);
+  // TODO(fzdob): this fails with EFAULT
```
Comment (cahirwpz) [inline bin/utest/cred.c:88]: @franciscozdo Why this function fails with EFAULT?

File: bin/utest/mmap.c [hunk 2]
```diff
@@ -56,53 +53,39 @@ static void mmap_with_hint(void) {
...
-  result = munmap(addr, 0x1000);
-  assert_ok(result);
+  syscall_ok(munmap(addr, 0x1000));
 
   /* munmapping again fails */
-  result = munmap(addr, 0x1000);
-  printf("result: %d\n", result);
-  assert(errno == EINVAL);
-  // assert_fail(result, EINVAL);
+  syscall_ok(munmap(addr, 0x1000));
```
Comment (cahirwpz) [inline bin/utest/mmap.c:72]: @franciscozdo Should it return `EINVAL`?

File: bin/utest/cred.c (hunk 1 above)
Comment (franciscozdo) [reply bin/utest/cred.c:88]: I can make an PR that fixes that return value for `getresuid` (and for `getresgid` where is the same problem).

File: bin/utest/mmap.c (hunk 2 above)
Comment (franciscozdo) [reply bin/utest/mmap.c:72]: No, that was mistake, and I forgot to remove that line when I was cleaning previous changes.

File: bin/utest/cred.c (hunk 1 above)
Comment (franciscozdo) [reply bin/utest/cred.c:88]: The manual entry I had referred to: https://www.freebsd.org/cgi/man.cgi?query=getresuid&sektion=2&n=1

Comment (cahirwpz) [reply bin/utest/cred.c:88]: Please fix it when this PR is merged. I'll leave TODO comment to be removed when the bug is fixed.

**Pull request merged**